dev/update_aggregate_docs.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
```

Comment:
I built the docs locally but the new pages didn't seem to appear in the table of contents

```shell
cd docs
./build.sh
```

![Screenshot 2024-10-01 at 1 07 22 PM](https://github.com/user-attachments/assets/693db8bd-dea2-4e5b-bf43-dcf26a03dbdb)



I had to manually run `dev/update_aggregate_docs.sh` 

And then when I rebuilt it says the new docs are not included in the TOC tree:

checking consistency... /Users/andrewlamb/Software/datafusion/docs/temp/user-guide/sql/aggregate_functions_new.md: WARNING: document isn't included in any toctree

So I added the new files to the toc tree:

```diff
diff --git a/docs/source/user-guide/sql/index.rst b/docs/source/user-guide/sql/index.rst
index 04d1fc228..4e02271b8 100644
--- a/docs/source/user-guide/sql/index.rst
+++ b/docs/source/user-guide/sql/index.rst
@@ -30,6 +30,7 @@ SQL Reference
    information_schema
    operators
    aggregate_functions
+   aggregate_functions_new
    window_functions
    scalar_functions
    sql_status
```

And now it shows up:

![Screenshot 2024-10-01 at 1 12 36 PM](https://github.com/user-attachments/assets/a47afd46-1778-4be6-821c-efb0655f7343)

Reply:
I wasn't thinking of making the generated docs public until they were much farther along ...

Reply:
That is fine too -- I also had this idea to incrementally expose the new docs but let's do it as a follow on PR.